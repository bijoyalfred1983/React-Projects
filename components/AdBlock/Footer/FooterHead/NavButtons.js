/* @flow */

import React from 'react';
import { Link } from 'react-router-dom';
import type { Element } from 'react';
import styles from './styles.scss';
import { jsTrack } from '../../../containers/Analytics/ClickTrack';

type Props = { btnContainer: Array<Object> };

export const trackFooterLink = (e, linkName) => {
  jsTrack.trackFooterNavMenu(linkName);
};

const NavButtons = ({ btnContainer }: Props): Element<'div'> => (
  <div className={styles.footerTop}>
    <ul className={styles.liWidthPrint}>
      {btnContainer.map(navbutton => (
        <li className={`col m4 s12 ${navbutton.className}`} key={navbutton.id}>
          <Link
            to={navbutton.linkTo}
            onClick={e => trackFooterLink(e, navbutton.content)}
          >
            <img src={navbutton.iconUrl} alt="not found" />
            <span>{navbutton.content.toUpperCase()}</span>
            <img
              src="/images/footer/right.svg"
              alt=""
              className={`hide-on-small-only ${styles.rightArrow}`}
            />
          </Link>
        </li>
      ))}
    </ul>
  </div>
);

export default NavButtons;
