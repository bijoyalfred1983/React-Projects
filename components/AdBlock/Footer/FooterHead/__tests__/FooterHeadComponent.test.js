/* @flow */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavButtons from '../NavButtons';
import FooterHeadComponent from '../index';
import { StaticRouter } from 'react-router-dom';

describe('<FooterHeadComponent />', () => {
  let PROPS;
  beforeEach(() => {
    PROPS = {
      btnContainer:  [
        {
          id: '1',
          linkTo: '/',
          content: 'locate a store',
          iconUrl: '/images/footer/location-pin.svg',
          className: ''
        },
        {
          id: '2',
          linkTo: '/trackOrder',
          content: 'track your order',
          iconUrl: '/images/footer/box.svg',
          className: ''
        }
      ]
    };
  });

  test('renders on FooterHeadComponent correctly', () => {
    const wrapper = mount(
      <StaticRouter context={{}}>
              <FooterHeadComponent {...PROPS} />
              </StaticRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});



