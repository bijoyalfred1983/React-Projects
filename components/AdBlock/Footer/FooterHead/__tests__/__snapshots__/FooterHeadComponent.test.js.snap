// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterHeadComponent /> renders on FooterHeadComponent correctly 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <FooterHeadComponent
      btnContainer={
        Array [
          Object {
            "className": "",
            "content": "locate a store",
            "iconUrl": "/images/footer/location-pin.svg",
            "id": "1",
            "linkTo": "/",
          },
          Object {
            "className": "",
            "content": "track your order",
            "iconUrl": "/images/footer/box.svg",
            "id": "2",
            "linkTo": "/trackOrder",
          },
        ]
      }
    >
      <div
        className="row"
      >
        <div
          className="footerHead"
        >
          <div
            className="col s12 m9 navBtnPrintWidth"
          >
            <NavButtons
              btnContainer={
                Array [
                  Object {
                    "className": "",
                    "content": "locate a store",
                    "iconUrl": "/images/footer/location-pin.svg",
                    "id": "1",
                    "linkTo": "/",
                  },
                  Object {
                    "className": "",
                    "content": "track your order",
                    "iconUrl": "/images/footer/box.svg",
                    "id": "2",
                    "linkTo": "/trackOrder",
                  },
                  Object {
                    "className": "hide-on-small-only",
                    "content": "we're hiring!",
                    "iconUrl": "/images/footer/pencil-write.svg",
                    "id": "3",
                    "linkTo": "/",
                  },
                ]
              }
            >
              <div
                className="footerTop"
              >
                <ul
                  className="liWidthPrint"
                >
                  <li
                    className="col m4 s12 "
                  >
                    <Link
                      onClick={[Function]}
                      replace={false}
                      to="/"
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        <img
                          alt="not found"
                          src="/images/footer/location-pin.svg"
                        />
                        <span>
                          LOCATE A STORE
                        </span>
                        <img
                          alt=""
                          className="hide-on-small-only rightArrow"
                          src="/images/footer/right.svg"
                        />
                      </a>
                    </Link>
                  </li>
                  <li
                    className="col m4 s12 "
                  >
                    <Link
                      onClick={[Function]}
                      replace={false}
                      to="/trackOrder"
                    >
                      <a
                        href="/trackOrder"
                        onClick={[Function]}
                      >
                        <img
                          alt="not found"
                          src="/images/footer/box.svg"
                        />
                        <span>
                          TRACK YOUR ORDER
                        </span>
                        <img
                          alt=""
                          className="hide-on-small-only rightArrow"
                          src="/images/footer/right.svg"
                        />
                      </a>
                    </Link>
                  </li>
                  <li
                    className="col m4 s12 hide-on-small-only"
                  >
                    <Link
                      onClick={[Function]}
                      replace={false}
                      to="/"
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        <img
                          alt="not found"
                          src="/images/footer/pencil-write.svg"
                        />
                        <span>
                          WE'RE HIRING!
                        </span>
                        <img
                          alt=""
                          className="hide-on-small-only rightArrow"
                          src="/images/footer/right.svg"
                        />
                      </a>
                    </Link>
                  </li>
                </ul>
              </div>
            </NavButtons>
          </div>
          <div
            className="col s12 m3 hide-on-small-only printIconWidth"
          >
            <IconComponent>
              <IconPane
                btnContainer={
                  Array [
                    Object {
                      "iconUrl": "/images/socialmedia/icon-facebook.svg",
                      "id": "1",
                      "url": "https://www.facebook.com/",
                    },
                    Object {
                      "iconUrl": "/images/socialmedia/icon-twitter.svg",
                      "id": "2",
                      "url": "https://www.twitter.com/",
                    },
                    Object {
                      "iconUrl": "/images/socialmedia/icon-you-tube.svg",
                      "id": "3",
                      "url": "https://www.youtube.com/",
                    },
                    Object {
                      "iconUrl": "/images/socialmedia/icon-instagram.svg",
                      "id": "4",
                      "url": "https://www.instagram.com/",
                    },
                    Object {
                      "iconUrl": "/images/socialmedia/icon-app-apple.svg",
                      "id": "5",
                      "url": "https://www.apple.com/",
                    },
                  ]
                }
              >
                <div
                  className="footerTop"
                >
                  <ul
                    className=""
                  >
                    <li>
                      <a
                        href="https://www.facebook.com/"
                      >
                        <img
                          alt=""
                          src="/images/socialmedia/icon-facebook.svg"
                        />
                      </a>
                    </li>
                    <li>
                      <a
                        href="https://www.twitter.com/"
                      >
                        <img
                          alt=""
                          src="/images/socialmedia/icon-twitter.svg"
                        />
                      </a>
                    </li>
                    <li>
                      <a
                        href="https://www.youtube.com/"
                      >
                        <img
                          alt=""
                          src="/images/socialmedia/icon-you-tube.svg"
                        />
                      </a>
                    </li>
                    <li>
                      <a
                        href="https://www.instagram.com/"
                      >
                        <img
                          alt=""
                          src="/images/socialmedia/icon-instagram.svg"
                        />
                      </a>
                    </li>
                    <li>
                      <a
                        href="https://www.apple.com/"
                      >
                        <img
                          alt=""
                          src="/images/socialmedia/icon-app-apple.svg"
                        />
                      </a>
                    </li>
                  </ul>
                </div>
              </IconPane>
            </IconComponent>
          </div>
        </div>
      </div>
    </FooterHeadComponent>
  </Router>
</StaticRouter>
`;
